### PR TITLE
[CON-550] Add nlog and fix issue reading encoding config

### PR DESCRIPTION
**Pull request by cofaulco**

### CODE REVIEW

File: src/SFA.DAS.Forecasting.Triggers/Startup.cs [hunk 1]
```diff
@@ -58,20 +62,60 @@ public IServiceProvider Build()
...
+                options.AddNLog(new NLogProviderOptions
+                {
+                    CaptureMessageTemplates = true,
+                    CaptureMessageProperties = true
+                });
+
+                options.AddConsole();
+                options.AddDebug();
+
+                nLogConfiguration.ConfigureNLog(Configuration);
```
Comment (lazcool) [inline src/SFA.DAS.Forecasting.Triggers/Startup.cs]: Looks like this line doesn't logically belong in the lambda. Could it be moved to the line after `AddLogging`?

File: src/SFA.DAS.Forecasting.Jobs.Infrastructure/Logging/NLogConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        }
+
+        private static void AddRedisTarget(LoggingConfiguration config, string appName)
+        {
+            var target = new RedisTarget
+            {
+                Name = "RedisLog",
+                AppName = appName,
+                EnvironmentKeyName = "EnvironmentName",
+                ConnectionStringName = "LoggingRedisConnectionString",
```
Comment (lazcool) [inline src/SFA.DAS.Forecasting.Jobs.Infrastructure/Logging/NLogConfiguration.cs]: Should `EnvironmentKeyName` and `ConnectionStringName` be set to string literals? Is there some sort of later substitution happening?

Comment (cofaulco) [reply src/SFA.DAS.Forecasting.Jobs.Infrastructure/Logging/NLogConfiguration.cs]: Yes it picks these from configuration

Comment (lazcool) [reply src/SFA.DAS.Forecasting.Jobs.Infrastructure/Logging/NLogConfiguration.cs]: Looks like the name of environment variables 😄

**Pull request merged**